pytorch_lightning/metrics/functional/classification.py

```diff
@@ -500,6 +504,12 @@ def auc(
         >>> auc(x, y)
         tensor(4.)
     """
+    rank_zero_warn(
+        "This `auc` was deprecated in v1.2.0 in favor of"
+        " `pytorch_lightning.metrics.functional.auc import auc`."
+        " It will be removed in v1.4.0", DeprecationWarning
+    )
+
     dx = x[1:] - x[:-1]
```

Comment:
can we internally call the new function so we do not need to maintain the code in 2 places till 1.4?

Reply:
Done :]